pkg/vsphere/disk/disk_manager_test.go

```diff
@@ -158,6 +158,26 @@ func TestCreateAndDetach(t *testing.T) {
 		if !assert.NoError(t, cerr) {
 			return
 		}
+		refs := child.attachedRefs.Count()
```

Comment:
how about a separate unit test for the reference counting? Use goroutines to get some parallelism.